src/compass/utils/range_split_spectrum.py

```diff
@@ -0,0 +1,173 @@
+
+    # Start block processing
+    lines_per_block = min(valid_length, lines_per_block)
+    num_blocks = int(np.ceil(valid_length / lines_per_block))
+
+    for band in range(in_ds.RasterCount):
+        for block in range(num_blocks):
+            line_start = block * lines_per_block
+            if block == num_blocks - 1:
+                block_length = valid_length - line_start
+            else:
+                block_length = lines_per_block
+
+            # Read a block of burst data
+            burst_data = in_ds.GetRasterBand(band+1).ReadAsArray(
+                burst.first_valid_sample,
+                burst.first_valid_line + line_start,
+                valid_width, block_length)
+
+            # Write extracted data
+            out_ds.GetRasterBand(band+1).WriteArray(burst_data, yoff=line_start)
+    out_ds.FlushCache()
+    return valid_burst_path
+
+
+def range_split_spectrum(burst, cfg_split_spectrum,
+                         scratch_path):
+    '''
+    Split burst range spectrum
+    Parameters:
+    ----------
+    burst: Sentinel1BurstSlc
+       S1-A/B burst object
+    cfg_split_spectrum: dict
+       Dictionary with split-spetrum options
+    scratch_path: str
+       Directory for storing temp files
+
+    Returns:
+    -------
+    burst_raster: isce3.io.Raster
+       3-bands ISCE3 Raster. Band #1: low band;
+       Band #2: main band; Band #3: high band
+    '''
+    length, width = burst.shape
+    lines_per_block = cfg_split_spectrum.lines_per_block
+
+    # In ISCE3, we can use raised cosine to implement S1-A/B Hamming
+    window_type = burst.range_window_type
+    window_type='Cosine' if window_type.casefold() == 'hamming' else window_type
+
+    # Extract bandwidths (bw) and create frequency vectors
+    half_bw = 0.5 * burst.range_bandwidth
+    low_bw = cfg_split_spectrum.low_band_bandwidth
+    high_bw = cfg_split_spectrum.high_band_bandwidth
+
+    low_freq_burst = burst.radar_center_frequency - half_bw
+    high_freq_burst = burst.radar_center_frequency + half_bw
+
+    low_band_freqs = np.array([low_freq_burst, low_freq_burst + low_bw])
+    high_band_freqs = np.array([high_freq_burst - high_bw, high_freq_burst])
+    low_center_freq = low_freq_burst + low_bw / 2
+    high_center_freq = high_freq_burst - high_bw / 2
+
+    # Initialize the split-spectrum parameter object. Note, constructor
+    # requires frequency (A/B) but this is not used anywhere below
+    rdr_grid = burst.as_isce3_radargrid()
+    split_spectrum_params = splitspectrum.SplitSpectrum(
+        rg_sample_freq=burst.range_sampling_rate,
+        rg_bandwidth=burst.range_bandwidth,
+        center_frequency=burst.radar_center_frequency,
+        slant_range=rdr_grid.slant_range,
+        freq='A')
+
+    # Save locally the valid part of the burst. Note, invalid data (zeros)
+    # cause issue when computing fft when splitting the range bandwidth
+    valid_burst_path = save_valid_burst(burst, lines_per_block, scratch_path)
```

Comment:
Couldn't we do this with VRT? I think it should be easily doable to write a VRT that points to the valid part of the burst.